tools/migration.py

```diff
@@ -0,0 +1,248 @@
+        return json.loads(response.text)
+
+    def get_from_old(self, service, path_param="", parameters=None):
+        """ Performs a GET call with the given information on the instance from
+        which the data should be migrated and returns the response as a JSON Object.
+        """
+        return self.get(self.old, service, path_param, parameters)
+
+    def get_from_new(self, service, path_param="", parameters=None):
+        """ Performs a GET call with the given information on the instance
+        to which the data should be migrated and returns the response as a JSON Object.
+        """
+        return self.get(self.new, service, path_param, parameters)
+
+    def put(self, service, data, path_param="", parameters=None):
+        """ Performs a PUT call from the client to the service with the given parameters and data.
+        The path parameter are additions to the path, e.g /service/id.
+        """
+        self.new.put(self.new.get_project_service_url(service) + path_param,
+                     parameters=parameters,
+                     json=data)
+
+    def get_matching_finding_id(self, finding_id):
+        """ Tries to find a matching finding in the new instance for
+        the given findings id of the old instance.
+        If no match could be found `None` is returned.
+        """
+        finding = self.get_from_old("findings-by-id", path_param=finding_id)
+        new_findings = self.get_from_new("findings",
+                                         path_param=finding["location"]["uniformPath"])
+        for new_finding in new_findings:
+            if self.match_finding(new_finding, finding):
+                return new_finding["id"]
+        return None
+
+    @staticmethod
+    def match_finding(finding1, finding2):
+        """ Checks if the given two findings are the same. """
+        if finding1["message"] != finding2["message"]:
+            return False
+
+        # some findings don't have a start line
+        has_line1 = "rawStartLine" in finding1["location"]
+        has_line2 = "rawStartLine" in finding2["location"]
+        if has_line1 != has_line2:
+            return False
+        if not has_line1:
+            return True
+        return finding1["location"]["rawStartLine"] == finding2["location"]["rawStartLine"]
+
+    @abstractmethod
+    def migrate(self):
+        pass
+
+
+class BlacklistMigrator(MigratorBase):
+    """ Class for migrating blacklists between two instances.
+    If some blacklisted finding cannot be found in the new instance, they will not
+    be migrated.
+    """
+
+    def migrate(self):
+        """ Migrates the blacklists. """
+        migrate_blacklist = self.get_blacklist_infos()
+        for blacklisted in migrate_blacklist:
+            new_id = self.get_matching_finding_id(blacklisted["findingId"])
+            if not new_id:
+                print("could not find finding %s in new Teamscale" % blacklisted["findingId"])
```

Comment:
if possible, please print not just finding IDs but links to the findings in error messages. that makes it easier for the user to look into issues
please mark error messages so they are easily identifed among the rest of the messages. E.g. use some logging framework